classes/class.memberorder.php

```diff
@@ -1097,4 +1096,53 @@ function get_test_order() {
 
 			return apply_filters( 'pmpro_test_order_data', $this );
 		}
+
+		/**
+		 * If a subscription does not already exist for this order, create one.
+		 *
+		 * @return bool True if a subscription was created, false if not.
+		 */
+		private function create_subscription_for_order() {
+			if ( empty( $this->subscription_transaction_id ) ) {
+				// No subscription transaction ID, so we don't need to create a subscription.
+				return false;
+			}
+
+			$get_subscription_args = array(
+				'subscription_transaction_id' => $this->subscription_transaction_id,
+				'gateway'                     => $this->gateway,
+				'gateway_environment'  	      => $this->gateway_environment,
+			);
+			$existing_subscription = PMPro_Subscription::get_subscription( $get_subscription_args );
+			if ( ! empty( $existing_subscription ) ) {
+				// We already have a subscription for this order, so we don't need to create a new one.
+				return false;
+			}
+
+			$create_subscription_args = array(
+				'user_id'                     => $this->user_id,
+				'membership_level_id'  		  => $this->membership_id,
+				'gateway'                     => $this->gateway,
+				'gateway_environment'  	      => $this->gateway_environment,
+				'subscription_transaction_id' => $this->subscription_transaction_id,
+			);
+			// Check if we have additional information from checkout to pass to the subscription.
```

Comment:
```suggestion
			// Check if we have additional information from checkout to pass to the subscription. If not set on checkout, this will use the current membership level set on the order.
```